app/donu/Main.hs

```diff
@@ -208,3 +200,20 @@ handleRequest r =
 
     succeed' :: (JS.ToJSON a, Show a) => a -> Snap.Handler x x Result
     succeed' = pure . succeed
+
+-- Driving Exposure -----------------------------------------------------------
+
+-- | Run the given exposure statements in the current session and return the
+-- resulting DisplayValues as a (JSON) list (or any error messages)
+stepExposureSession :: [Exposure.Stmt] -> Snap.Handler DonuApp DonuApp Result
+stepExposureSession stmts =
+  do env <- Snap.with exposureSessions $
+       fromMaybe Exposure.initialEnv <$> getExposureSessionState
+     res <- liftIO $ Exposure.evalStmts stmts env
```

Comment:
This isn't your fault, but it occurs to me that we have two functions—`evalStmts` and `evalLoop`—which do the exact same thing. I wonder if we should converge on one of them and delete the other.